x-pack/plugins/security_solution/public/overview/components/detection_response/hooks/use_navigate_to_timeline.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useDispatch } from 'react-redux';
+
+import { getDataProvider } from '../../../../common/components/event_details/table/use_action_cell_data_provider';
+import { sourcererActions } from '../../../../common/store/sourcerer';
+import { SourcererScopeName } from '../../../../common/store/sourcerer/model';
+import { DataProvider, TimelineId, TimelineType } from '../../../../../common/types/timeline';
+import { useCreateTimeline } from '../../../../timelines/components/timeline/properties/use_create_timeline';
+import { updateProviders } from '../../../../timelines/store/timeline/actions';
+
+export const useNavigateToTimeline = () => {
+  const dispatch = useDispatch();
+
+  const clearTimeline = useCreateTimeline({
+    timelineId: TimelineId.active,
+    timelineType: TimelineType.default,
+  });
+
+  const navigateToTimeline = (dataProvider: DataProvider) => {
```

Comment:
I actually talked to @janmonschke about this solution. There is some background refactoring on investigations team to allow for this ability. I got the impression that down the road this would be more streamlined, but for now to use what I have here. Is that fair Jan?

Reply:
Spoke to Jan on this in zoom, Kevin would be the person to talk to about a common implementation or incoming changes to this ability. Unfortunately Kevin is also out til next week. I will resolve this for now, and chat with him when he's back to see if this is possible as a future ticket or if I should wait til his work is complete.